periodic sync upstream KF to midstream ODH

tarilabs opened the pull request:
content:
- keep https://github.com/kubeflow/model-registry/pull/37
- keep https://github.com/kubeflow/model-registry/pull/38
- keep https://github.com/kubeflow/model-registry/pull/29
- skip https://github.com/kubeflow/model-registry/pull/36
by revert with e9167d92e93eeea8d6c555c30e20705c99eddc82
- keep https://github.com/kubeflow/model-registry/pull/40

dhirajsb reviewed (approved):
nice catch :laughing:

dependabot[bot] committed:
build(deps): bump google.golang.org/protobuf from 1.32.0 to 1.33.0 (#37)

Bumps google.golang.org/protobuf from 1.32.0 to 1.33.0.

---
updated-dependencies:
- dependency-name: google.golang.org/protobuf
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump google.golang.org/grpc from 1.62.0 to 1.62.1 (#38)

Bumps [google.golang.org/grpc](https://github.com/grpc/grpc-go) from 1.62.0 to 1.62.1.
- [Release notes](https://github.com/grpc/grpc-go/releases)
- [Commits](https://github.com/grpc/grpc-go/compare/v1.62.0...v1.62.1)

---
updated-dependencies:
- dependency-name: google.golang.org/grpc
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

antisaling committed:
OAS: fix minor issues and bump version (#29)

* OAS: fixup ID -> Id in parameter names

... to preserve camelCase.

Signed-off-by: Isabella Basso do Amaral <idoamara@redhat.com>

* fix REST api to use ModelVersionUpdate in PATCH method

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

* bump OAS version to v1alpha3

Signed-off-by: Isabella Basso do Amaral <idoamara@redhat.com>

---------

Signed-off-by: Isabella Basso do Amaral <idoamara@redhat.com>
Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>
Co-authored-by: Matteo Mortari <matteo.mortari@gmail.com>

tarilabs committed:
add testing of KF MR Manifests (#36)

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

dependabot[bot] committed:
build(deps): bump google.golang.org/protobuf from 1.32.0 to 1.33.0 in /csi (#40)

Bumps google.golang.org/protobuf from 1.32.0 to 1.33.0.

---
updated-dependencies:
- dependency-name: google.golang.org/protobuf
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

tarilabs committed:
Revert skip "add testing of KF MR Manifests (kubeflow#36)"

This reverts commit aeeffc0ac4bf44f4d443e0d9f36f3e66063c2c5c.

tarilabs committed:
Merge branch 'odh-main' into tarilabs-20240318-midstreamsync

rareddy commented on csi/GET_STARTED.md:
````diff
@@ -82,7 +82,7 @@ Then, in the same terminal where you exported `MR_HOSTNAME`, perform the followi
 ```bash
 curl --silent -X 'POST' \
-  "$MR_HOSTNAME/api/model_registry/v1alpha2/registered_models" \
+  "$MR_HOSTNAME/api/model_registry/v1alpha3/registered_models" \
````
I know that we want to be back-compatible, but we do not have yet any stable released versions should we be changing the version numbers this aggressively? may be not right issue to ask this question either :(

rareddy replied:
I recommend that if there are any API changes we update the version number only once for the given release. I am sure you might be already doing that, just wanted to understand.

tarilabs replied:
Yes, we can avoid "releasing too often", at the same time I understood we wanted to be strict on bumping the version number for every breaking change nevertheless.

Here is the short-blanket: either we minimize releases keeping the number bumps, or we release as often as needed and we relax strictness on the backward incompatibility of the REST API number. my2c.

rareddy commented:
/lgtm
